python/doc/examples/numerical_methods/optimization/plot_optimization_pagmo.py

```diff
@@ -0,0 +1,71 @@
+"""
+Multi-objective optimization using Pagmo
+========================================
+"""
+# %%
+# In this example we are going to explore optimization using OpenTURNS' `Pagmo <https://esa.github.io/pagmo2/>`_ interface.
+
+# %%
+from __future__ import print_function
+import openturns as ot
+from openturns.viewer import View
+ot.Log.Show(ot.Log.NONE)
+
+# %%
+# List available algorithms
+for algo in ot.Pagmo.GetAlgorithmNames():
+    print(algo)
+
+# %%
+# More details on Pagmo algorithms are available `here <https://esa.github.io/pagmo2/overview.html#list-of-algorithms>`_ .
+
+# %%
+# Create the problem, from ZDT test suite
+f = ot.SymbolicFunction(['x1', 'x2'], ['x1', 'var g := (1.0+9*x2); g * (1.0 - sqrt(x1 / g))'])
+zdt1 = ot.OptimizationProblem(f)
+zdt1.setBounds(ot.Interval([0.0] * 2, [1.0] * 2))
+
+# %%
+# Create the initial population
+pop0 = ot.ComposedDistribution([ot.Uniform(0.0, 1.0)] * 2).getSample(100)
+
+# %%
+# Create the algorithm
+algo = ot.Pagmo(zdt1, 'nsga2', pop0)
+algo.setGenerationNumber(250)
+
+# %%
+# Benefit from parallel evaluations if the function allows it
+algo.setBlockSize(8)
+
+# %%
+# Run the algo
+algo.run()
+pop1 = algo.getResult().getFinalPoints()
+
+# %%
+# Show the first Pareto fronts
+fronts = algo.getResult().getParetoFronts()
+len(fronts)
+
+# Show the first Pareto fronts
+graph = ot.Graph("Fronts", "X1", "X2", True, 'topright')
+palette = ot.Drawable.BuildDefaultPalette(len(fronts))
+for k in range(2):
+    front = fronts[k].sortAccordingToAComponent(0)
+    print(front)
+    data = ot.Sample(2*front.getSize()-1, 2)
+    for i in range(front.getSize()):
+        data[2*i] = front[i]
+        if i!= front.getSize()-1:
+            data[2*i+1,0] = front[i,0]
+            data[2*i+1,1] = front[i+1,1]
+    curve = ot.Curve(data)
+    curve.setColor(palette[k])
+    curve.setLegend(f"front {k}")
+    graph.add(curve)
+graph.setGrid(True)
+View(graph)
```

Comment:
This produces:
![image](https://user-images.githubusercontent.com/31351465/155174959-4c53bf3b-670c-4dde-a534-99ab308cd6e3.png)

This is a trivial Pareto front: I suggest to present a more significant example.

Reply:
I get two fronts, see the uploaded picture

Reply:
The new picture is indeed a little better:
![image](https://user-images.githubusercontent.com/31351465/155417851-75fb772d-b940-4bb3-978d-2ad781f47f83.png)

But this is not that motivating. 
Please consider the examples from the Python interface Pygmo to Pagmo:
https://esa.github.io/pygmo/tutorials/adding_a_new_optimization_problem.html
https://esa.github.io/pygmo/tutorials/basic_mo_functionalities.html
The first example produces:
![image](https://user-images.githubusercontent.com/31351465/155418051-3974dd2d-f029-45ad-84c7-937a058ef5fb.png)
The second example produces the Pareto front for 10 different generations:
![image](https://user-images.githubusercontent.com/31351465/155418111-479f9782-e607-499c-b26a-1f429a88f3be.png)
That would be more meaningful for users.